File: retos-python/07_ataque_grinch.py
# Resultado 5 ⭐
def fixPackages(packages):
    tmpPack = packages

    while '(' in tmpPack:
        startIndex = tmpPack.rfind('(')
        endIndex = tmpPack.find(')',startIndex)
        
        newPack = tmpPack[startIndex:endIndex+1]
        packRev = ''.join(list(newPack)[::-1][1:-1])

        tmpPack = tmpPack.replace(newPack,packRev)

    return tmpPack

File: retos-ts/07_ataque_grinch.ts
// Resultado 5 ⭐
function fixPackages(packages: string): string {
    let tmpPack: string = packages;

    while (tmpPack.includes('(')) {
        const startIndex: number = tmpPack.lastIndexOf('(');
        const endIndex: number = tmpPack.indexOf(')',startIndex);

        const newPack: string = tmpPack.substring(startIndex,endIndex + 1);
        const reverse: string = [...newPack].slice(1,-1).reverse().join("")
        tmpPack = tmpPack.replace(newPack,reverse);

    } 

    return tmpPack;
}